src/mono/browser/runtime/marshal-to-js.ts

```diff
@@ -59,7 +59,7 @@ export function initialize_marshalers_to_js(): void {
 }
 
 export function bind_arg_marshal_to_js(sig: JSMarshalerType, marshaler_type: MarshalerType, index: number): BoundMarshalerToJs | undefined {
-    if (marshaler_type === MarshalerType.None || marshaler_type === MarshalerType.Void) {
+    if (marshaler_type === MarshalerType.None || marshaler_type === MarshalerType.Void || marshaler_type === MarshalerType.Discard) {
```

Comment:
What is the role of `marshaler_type === MarshalerType.Discard`? We didn't have it originally.

Reply:
It means that the result value doesn't have to be marshaled back. 
Optimization, we had it for long time, but here we didn't take advantage of it.